tests/rptest/tests/debug_bundle_test.py

```diff
@@ -0,0 +1,189 @@
+                               self.redpanda,
+                               self.topic,
+                               group="debug-bundle-group",
+                               num_msgs=msg_count,
+                               auth_tuple=self.redpanda.SUPERUSER_CREDENTIALS
+                               if use_basic_auth else None)
+        consumer.start()
+
+        wait_until(lambda: consumer.message_count > msg_count / 2,
+                   timeout_sec=30,
+                   backoff_sec=1)
+
+        node = self.redpanda.started_nodes()[0]
+        default_bundle_name = "debug-bundle.zip"
+
+        admin = Admin(self.redpanda)
+        journalctl_date = datetime.now().strftime("%Y-%m-%d")
+        res = admin.start_debug_bundle(node=node,
+                                       logs_since=journalctl_date,
+                                       logs_size_limit="100MiB",
+                                       metrics_interval="10s",
+                                       auth=auth_tuple)
+        assert res.status_code == requests.codes.accepted
+
+        try:
+            admin.start_debug_bundle(node=node)
+        except HTTPError as ex:
+            # Expect HTTP Error 429 since a bundle is already running
+            if ex.response.status_code == requests.codes.too_many_requests:
+                pass
+            else:
+                raise
+
+        def bundle_status():
+            try:
+                res = admin.debug_bundle_status(filename=default_bundle_name,
+                                                node=node,
+                                                auth=auth_tuple)
+                # OK status means the bundle is available for download
+                return res.status_code == requests.codes.ok
+            except HTTPError as ex:
+                # Not found status means the bundle is in progress but not yet ready
+                if ex.response.status_code == requests.codes.not_found:
+                    return False
+                # Gone status means the bundle is not in progress or it is not on disk
+                elif ex.response.status_code == requests.codes.gone:
+                    return False
+                else:
+                    raise
+
+        wait_until(bundle_status, timeout_sec=120, backoff_sec=10)
+
+        def download_bundle(local_bundle_filename: str, n: ClusterNode,
+                            auth_tuple: tuple):
+            res = admin.download_debug_bundle(filename=default_bundle_name,
+                                              node=n,
+                                              auth=auth_tuple)
+            assert res.status_code == requests.codes.ok
+            with open(local_bundle_filename, "wb") as local_bundle:
+                local_bundle.write(res.content)
+
+            assert os.path.isfile(local_bundle_filename)
+
+        # Download the bundle twice so we can compare the contents.
+        # The file will be on the orchestrator node
```

Comment:
Just wondering, is it possible in ducktape to transfer a file using `scp` or something?  If it's possible, I think it would be a better test to grab that file and compare it with the downloaded one to really verify that the downloaded file matches the generated one.

Reply:
Ducktape's internal `sftp_client` is an instance of [Paramiko's SFTP client](https://docs.paramiko.org/en/stable/api/sftp.html) and that has methods to fetch a remote file to the local disk. I will use that in the next push.